src/components/Tool/FixMailingAddresses/Contact.tsx

```diff
@@ -234,33 +206,40 @@ const Contact: React.FC<Props> = ({
                 </Box>
               </Grid>
             </Hidden>
-            {addresses.map((address) => (
-              <Fragment key={address.id}>
-                <Grid item xs={12} md={5} className={classes.paddingB2}>
-                  <Box display="flex" justifyContent="space-between">
-                    <Grid item md={8}>
-                      <Hidden mdUp>
+            {addressesData &&
+              addressesData.map((address) => (
+                <Fragment key={address.id}>
```

Comment:
Could you use optional changing here as it makes it easier to read.
```suggestion
            {{addressesData?.map((address) => (
                <Fragment key={address.id}>
```